patches/tModLoader/Terraria/MessageBuffer.cs.patch

```diff
@@ -361,6 +361,14 @@
  							tileEntity.ID = num48;
  							TileEntity.ByID[tileEntity.ID] = tileEntity;
  							TileEntity.ByPosition[tileEntity.Position] = tileEntity;
+@@ -3304,6 +_,7 @@
+ 						obj.luckPotion = luckPotion;
+ 						obj.HasGardenGnomeNearby = hasGardenGnomeNearby;
+ 						obj.RecalculateLuck();
++						PlayerHooks.ModifyLuck(obj, ref obj.luck); //Neccesary? Should this be in PlayerIO instead?
```

Comment:
I did that initially, but I didn't like how returning PreModifyLuck false also made it so ModifyLuck didn't do anything (since it cancelled out all of RecalculateLuck), so I put it outside the method.